src/RevisionPopupWidget.js

```diff
@@ -25,48 +25,51 @@ OO.inheritClass( RevisionPopupWidget, OO.ui.PopupWidget );
 /**
  * Get markup for the diff size.
  * @param {number} size
- * @return {string}
+ * @return {jQuery}
  */
 function getSizeHtml( size ) {
-	let sizeClass;
+	const $diffBytes = $( '<span>' );
+	$diffBytes.addClass( 'mw-diff-bytes' );
 
+	let sizeClass;
 	if ( size > 0 ) {
 		sizeClass = 'mw-plusminus-pos';
 	} else if ( size < 0 ) {
 		sizeClass = 'mw-plusminus-neg';
 	} else {
 		sizeClass = 'mw-plusminus-null';
 	}
+	$diffBytes.addClass( sizeClass );
 
-	return `
-		<span class="${sizeClass} mw-diff-bytes">` +
-			`${size > 0 ? '+' : ''}${mw.language.convertNumber( size )}` +
-		'</span>';
+	$diffBytes.text( `${size > 0 ? '+' : ''}${mw.language.convertNumber( size )}` );
+	return $diffBytes;
 }
 
 /**
  * Get markup for the edit summary.
  * @param {Object} data As returned by Api.prototype.getTokenInfo().
- * @return {string}
+ * @return {jQuery|null}
  */
 function getCommentHtml( data ) {
+	const $revCommentDiv = $( '<div>' );
+	$revCommentDiv.addClass( 'wwt-revisionPopupWidget-comment' );
+
 	if ( data.comment === '' ) {
 		// No edit summary.
-		return '';
+		return null;
 	} else if ( data.comment === undefined ) {
 		// Not yet available.
-		return `
-			<div class="wwt-revisionPopupWidget-comment">
-				<div class="wwt-shimmer www-shimmer-animation"></div>
-				<div class="wwt-shimmer www-shimmer-animation"></div>
-			</div>`;
+		const $shimmerDiv = $( '<div>' );
+		$shimmerDiv.addClass( 'wwt-shimmer www-shimmer-animation' );
+		$revCommentDiv.append( $shimmerDiv, $shimmerDiv );
+	} else {
+		$revCommentDiv.addClass( 'wwt-revisionPopupWidget-comment-transparent' );
+		const $commentSpan = $( '<span>' );
+		$commentSpan.addClass( 'comment comment--without-parentheses wwt-revisionPopupWidget-comment' );
+		$commentSpan.append( Tools.bidiIsolate( data.comment ) );
```

Comment:
Also, `Tools.bidiIsolate` needs to be re-fortified. Right now, it automatically html's any comment we get, which isn't great.

It should go back to only outputting literal strings as .text() which forces the callers to demonstrate the safety of sent data, like the API comment.

Reply:
Sorry for the mess I created! I just wanted to note that the edit summary can contain internal links and we'll want to render that HTML. Whether that works with `.text()` I'm not sure.

Reply:
Okay, two things here, following up [from the ticket](https://phabricator.wikimedia.org/T232802#5558253) --

1. Demonstrable security: By making the tool *itself* force people to explicitly send it jQuery objects when they intend to do so, it makes sure that those choices are not only intentional, but also that they *look* intentional to anyone reading the code.
2. With the current behavior, we cannot send a raw text that we want to *not* parse as html, ever, which is bad. An example would be a username; we actually never want to html-parse a username, and the fact that it might be safe(ish) 90% of the time doesn't mean we shouldn't fortify against it. 

When you look at the code, this:

```
Tools.bidiIsolate( $( '<span>' ).html( data.comment ) );
```
Tells you that we intentionally html-parse comments, and this --

```
Tools.bidiIsolate( data.username );
```
Tells the code reader that they don't need to worry what username has inside it, whether it's justified or not.

We shouldn't underestimate perceived security, and we should fortify ourselves against html-parsing things we really shouldn't.

(BTW, while it's true that usernames *should* be safe, it's also worth noting that the array of illegal characters was somewhat slowly added to Wikipedia, and we never actually removed past usernames that used them, which means we might actually have situations where usernames *are not* html-safe.)

Reply:
Of course, good points. I've switched it to using `.text()` for strings, and appending all else.